cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java

```diff
@@ -168,6 +184,29 @@ private CloudCredentialStatus verifyAccessKeySecretKeyIsAssumable(CloudCredentia
         return credentialStatus;
     }
 
+    private void checkRoleIsAssumableWithoutExternalId(CredentialVerificationContext credentialVerificationContext, AwsCredentialView awsCredential) {
+        if (credentialVerificationContext.getCreationVerification()) {
+            String roleArn = awsCredential.getRoleArn();
+            try {
+                credentialClient.retrieveSessionCredentialsWithoutExternalId(awsCredential);
+                String message = String.format("CDP Control Pane is able to perform 'sts:AssumeRole' on '%s' role without external id. " +
+                        "The role is vulnerable and could be exploited by confused deputy attacks. " +
+                        "Please update the role's trust relationship to have a condition on external id " +
+                        "or re-create the role with external id configured in the create role wizard.", roleArn);
+                LOGGER.warn(message);
+                throw new AwsPossibleConfusedDeputyException(message);
+            } catch (AmazonClientException ae) {
+                if (ae.getMessage().contains(ROLE_IS_NOT_ASSUMABLE_ERROR_MESSAGE_INDICATOR)) {
+                    String msg = String.format("Consider the specified role as secured. " +
+                            "CDP Control Pane is not authorized to perform sts:AssumeRole on '%s' role without external Id.", roleArn);
```

Comment:
I think here the problem is not the missing external id right ?

Reply:
That's a happy path and only a log is called to leave a sign about it.